thirdparty/github.go

```diff
@@ -1765,6 +1765,81 @@ func GetMergeablePullRequest(ctx context.Context, issue int, githubToken, owner,
 	return pr, nil
 }
 
+// HasEvergreenBranchProtections checks if any branch protection rule begins "evergreen".
+// We deliberately check for the prefix because users may have installed variant-level branch protections.
+func HasEvergreenBranchProtection(ctx context.Context, token, owner, repo, branch string) (bool, error) {
+	rules, err := GetEvergreenBranchProtectionRules(ctx, token, owner, repo, branch)
+	if err != nil {
+		return false, errors.Wrap(err, "getting branch protection rules")
+	}
+	for _, rule := range rules {
```

Comment:
GetEvergreenBranchProtectionRules already checks the Evergreen prefix, and so the only rules returned here are those. 

That being said, I have a preference for keeping these third party functions really simple -- they make the Github client, call the Github function and handle it, and return the result. Then we can use the result however we want (in helper functions within thirdparty and outside, so in this function we could keep the logic of iterating through the rules). If we were to do that, then we could also call this hypothetical GetBranchProtection function in the Install function, and then just loop through the checks.

Reply:
True. I think in that case we can remove this function entirely. Removed.